2024-05-15 Change log: Introduce `fern-bot` generated change log (#3747)
astronomer/docs: api/fern/apis/platform/changelog/2024-05-15.md

## Summary

> This update introduces new endpoints for managing your deploys for a Deployment. To run your code on Astro, you need to deploy it to a Deployment. You can deploy part or all of an Astro project to an Astro Deployment. [Learn more.](https://docs.astronomer.io/astro/deploy-code)

Reviewer: ```suggestion

This update introduces new endpoints for managing deploys, including image deploys, DAG deploys, and deploy rollbacks. See [Astro documentation](https://docs.astronomer.io/astro/deploy-code) for more information.
```

### Added

New endpoints:
  - `GET /organizations/{organizationId}/deployments/{deploymentId}/deploys` to list deploys for a deployment.
  - `POST /organizations/{organizationId}/deployments/{deploymentId}/deploys` to create a new deploy.
  - `GET /organizations/{organizationId}/deployments/{deploymentId}/deploys/{deployId}` to retrieve a specific deploy.
  - `POST /organizations/{organizationId}/deployments/{deploymentId}/deploys/{deployId}` to update an existing deploy.

>   - `POST /organizations/{organizationId}/deployments/{deploymentId}/deploys/{deployId}/finalize` to finalize the deployment of a deploy.

Reviewer: ```suggestion
  - `POST /organizations/{organizationId}/deployments/{deploymentId}/deploys/{deployId}/finalize` to finalize a deploy.
```

>   - `POST /organizations/{organizationId}/deployments/{deploymentId}/deploys/{deployId}/rollback` to rollback a deploy.

Reviewer: ```suggestion
  - `POST /organizations/{organizationId}/deployments/{deploymentId}/deploys/{deployId}/rollback` to roll back a deploy.
```
